### PR TITLE
feat: Add sections to `key-values` components

### PR DESCRIPTION
**Pull request by beneditomauro**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-robin-revamp/issues/330 |
| Decisions | - Add a section property with a `key` that will be the name of the sections and with its own key values |
| Animated GIF | gLM7WLkhth |

### CODE REVIEW

File: react/components/molecules/key-values/key-values.js [hunk 1]
```diff
@@ -156,6 +156,36 @@ export class KeyValues extends PureComponent {
                                 {item.valueComponent}
                             </KeyValue>
                         )}
+                        {item.section && (
+                            <View>
+                                <Text style={styles.sectionTitle}>{item.key}</Text>
```
Comment (3rdvision) [inline react/components/molecules/key-values/key-values.js:161]: this seems to be duplicate just to show the section... can't we just add that condition for <Text> like:
```suggestion
                                {item.section && <Text style={styles.sectionTitle}>{item.key}</Text>}
```

and above if you want a different style make that condition inside the style?
this would avoid duplicating this code

File: react/components/molecules/key-values/key-values.stories.js [hunk 2]
```diff
@@ -12,7 +12,21 @@ storiesOf("Molecules", module)
...
+                    { key: "section-Company", value: "Platforme", border: "none" },
+                    {
+                        key: "section-Position",
+                        value: "Head of Software Development",
+                        border: "soft"
+                    },
+                    { key: "section-Birth date", value: "14/03/1993", border: "hard" },
+                    { key: "section-Nationality", value: "Portuguese" }
+                ]
+            }
```
Comment (3rdvision) [inline react/components/molecules/key-values/key-values.stories.js:29]: can't we simplify this structure by simply adding separator label objects in the position where we want it to show instead of making this section like structure? Example:
```suggestion
            { separator: "Nationality" },
            { key: "E-mail", value: "gcc@platforme.com" },
            { key: "Company", value: "Platforme", border: "none" },
            { key: "Position", value: "Head of Software Development" },
            { key: "Birth Date", value: "14/03/1993" },
            { key: "nationality", value: "Portuguese" }
```

Also I'm starting to think that we could simplify even more by creating a component called `SeparatorTitle` or similar and use it in robin just before the key-values that we wanted to section. It would be simpler to implement and avoid possible regressions. What do you think is best?

**Review by 3rdvision — CHANGES_REQUESTED** [review]

File: react/components/molecules/key-values/key-values.js [hunk 3]
```diff
@@ -78,7 +78,7 @@ export class KeyValues extends PureComponent {
     };
 
     _shouldShow = item => {
-        return Boolean(item.value) || this.props.showUnset;
+        return Boolean(item.value || this.props.showUnset) && !item.section;
```
Comment (3rdvision) [inline react/components/molecules/key-values/key-values.js:81]: is this change still needed?

File: react/components/molecules/key-values/key-values.stories.js [hunk 4]
```diff
@@ -12,7 +12,21 @@ storiesOf("Molecules", module)
             { key: "Company", value: "Platforme", border: "none" },
             { key: "Position", value: "Head of Software Development", border: "soft" },
             { key: "Birth date", value: "14/03/1993", border: "hard" },
-            { key: "Nationality", value: "Portuguese" }
+            { key: "Nationality", value: "Portuguese" },
```
Comment (3rdvision) [inline react/components/molecules/key-values/key-values.stories.js:15]: this needs to be updated to use the new approach, right?

File: react/components/molecules/key-values/key-values.js (hunk 3 above)
Comment (beneditomauro) [reply react/components/molecules/key-values/key-values.js:81]: yes, it's either that or change the logic of the template, I think this is the best option

**Review by 3rdvision — APPROVED** [review]

**Review by joao-conde — APPROVED** [review]

**Review by joamag — APPROVED** [review]
